packages/sidecar/src/cli/run.ts

```diff
@@ -145,19 +148,30 @@ logger.error(`The port ${port} might already be in use — most likely by anothe
   }
   const cmdStr = cmdArgs.join(" ");
   logger.info(`Starting command: ${cmdStr}`);
+  // When we have Docker Compose override YAML (for -f -), we need to pipe stdin
+  // Otherwise, inherit stdin to relay user input to the downstream process
+  const stdinMode: "pipe" | "inherit" = dockerComposeOverride ? "pipe" : "inherit";
   const runCmd = spawn(cmdArgs[0], cmdArgs.slice(1), {
     cwd: process.cwd(),
     env,
     shell,
     windowsVerbatimArguments: true,
     windowsHide: true,
-    // We `inherit` the stdin so we relay any user input to the downstream process
-    // This also helps sending the SIGINT signal directly. Alternative would be to
-    // use "pipe" and call `runCmd.stdin.end()` to signal the end of the input stream.
-    // Never ever use `ignore` as it keeps stdin open forever, preventing a graceful shutdown
-    stdio: ["inherit", "pipe", "pipe"],
+    stdio: [stdinMode, "pipe", "pipe"],
   });
 
+  const { stdout, stderr } = runCmd;
+  if (!stdout || !stderr) {
+    logger.error("Failed to create process streams");
+    process.exit(1);
+  }
+
+  // If we have Docker Compose override YAML, write it and close stdin
+  if (dockerComposeOverride && runCmd.stdin) {
+    runCmd.stdin.write(dockerComposeOverride);
+    runCmd.stdin.end();
+  }
```

Comment:
If we have `dockerComposeOverride` but not `runCmd.stdin` we should throw. (we can either simply remove the `runCmd.stdi` check here and let it throw if it's missing or handle this case explicitly)